runtime/src/signed_extensions/check_call_allowed.rs

```diff
@@ -0,0 +1,148 @@
+    pub fn new() -> Self {
+        Self(sp_std::marker::PhantomData)
+    }
+}
+
+impl sp_std::fmt::Debug for CheckCallAllowed<Runtime> {
+    #[cfg(feature = "std")]
+    fn fmt(&self, f: &mut sp_std::fmt::Formatter) -> sp_std::fmt::Result {
+        write!(f, "CheckCallAllowed")
+    }
+
+    #[cfg(not(feature = "std"))]
+    fn fmt(&self, _: &mut sp_std::fmt::Formatter) -> sp_std::fmt::Result {
+        Ok(())
+    }
+}
+
+impl Default for CheckCallAllowed<Runtime> {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl CheckCallAllowed<Runtime> {
+    // Checks to see if the account has any rivalrous locks that would conflict
+    // with the FRAME staking pallet lock
+    fn has_no_conflicting_locks(who: &AccountId) -> bool {
+        let existing_locks = Balances::locks(who);
+
+        if existing_locks.len() == 0 {
+            return true;
+        }
+
+        if existing_locks.len() == 1 && existing_locks[0].id == STAKING_LOCK_ID {
+            return true;
+        }
+
+        let existing_lock_ids: Vec<LockIdentifier> =
+            existing_locks.iter().map(|lock| lock.id).collect();
+
+        !Runtime::are_locks_conflicting(&STAKING_LOCK_ID, existing_lock_ids.as_slice())
+    }
+
+    // Recursively checks calls until one invalid bonding call is detected and returns false,
+    // otherwise returns true.
+    fn has_no_invalid_bonding_calls(who: &AccountId, calls: Vec<Call>) -> bool {
+        let mut all_calls_valid = true;
+
+        for call in calls.into_iter() {
+            all_calls_valid = all_calls_valid
+                && match call {
+                    // Calls that can contain other Calls and must be checked recursively..
+                    Call::Utility(substrate_utility::Call::<Runtime>::batch { calls }) => {
+                        Self::has_no_invalid_bonding_calls(who, calls.to_vec())
+                    }
+                    Call::Utility(substrate_utility::Call::<Runtime>::as_derivative {
+                        call,
+                        ..
+                    }) => Self::has_no_invalid_bonding_calls(who, vec![*call]),
+
+                    // Bonding
+                    Call::Staking(pallet_staking::Call::<Runtime>::bond { .. }) => {
+                        Self::has_no_conflicting_locks(who)
+                    }
+
+                    // should we prevent Sudo from bypassing these checks?
+                    // Call::Sudo(pallet_sudo::Call::<Runtime>::sudo { call }) => ...
+                    // Call::Sudo(pallet_sudo::Call::<Runtime>::sudo_as { who, call }) => ...
+                    // Call::Sudo(pallet_sudo::Call::<Runtime>::sudo_unchecked_weight { call }) => ...
+                    // Call::Utility(substrate_utility::Call::<Runtime>::batch_all { calls }) => ...
+                    // Call::Utility(substrate_utility::Call::<Runtime>::force_batch { calls }) => ...
+                    // Call::Utility(substrate_utility::Call::<Runtime>::dispatch_as { calls }) => ...
```

Comment:
The list should also include `multisig.asMulti` and `multisig,asMultiThreshold1`

Reply:
will add